torch/utils/data/datapipes/utils/common.py

```diff
@@ -32,6 +32,8 @@ def onerror(err : OSError):
         raise err
 
     for path, dirs, files in os.walk(root, onerror=onerror):
+        dirs.sort()  # Make os.walk traverse directories in sorted order
```

Comment:
Auch, this is shady, you are inplace modifying internal list of the os.walk function. It work only because this line https://github.com/python/cpython/blob/f4c03484da59049eb62a9bf7777b963e2267d187/Lib/os.py#L407 doesn't do shallow copy before return. I suggest to put this info into comment. Alternatively make a copy of os.walk function locally to freeze state.

Reply:
@VitalyFedyunin @ejguan We ended up closing this PR but this one https://github.com/pytorch/pytorch/pull/70435/ does something similar.

Reply:
I agree to leave it as is, but please add inline code comment. I don't want future you or future me spending hours to figure out why it is not working any more =)

Reply:
Will do!